### PR TITLE
Fix applyAnimation() scale when original scale is 0

### PR DESCRIPTION
**Pull request by iamlkj**
Fix wrong orientation when calling applyAnimation() with node scale close to 0. Originally it'll still compute orientation when scale is 0 which lead to a wrong initial quaternion(0.5,0,0,0). The fix gives default quaternion if scale is 0

### CODE REVIEW

File: libs/gltfio/src/math.h [hunk 1]
```diff
@@ -75,10 +75,13 @@ inline void decomposeMatrix(const filament::math::mat4f& mat, filament::math::fl
         clone[0] /= s.x;
         clone[1] /= s.y;
         clone[2] /= s.z;
+        // Extract rotation.
+        *rotation = clone.toQuaternion();
+    }
```
Comment (romainguy) [inline libs/gltfio/src/math.h:80]: Looks good to me, just a simple formatting issue. This should written:
```
} else {
    *rotation = // ... note the 4 spaces indentation
}
```

File: libs/gltfio/src/math.h [hunk 2]
```diff
@@ -75,10 +75,13 @@ inline void decomposeMatrix(const filament::math::mat4f& mat, filament::math::fl
...
+    else {
+      // Set to idendity if close to zero
+      *rotation = filament::math::quatf(1);
```
Comment (prideout) [inline libs/gltfio/src/math.h:83]:
```suggestion
      *rotation = quatf(1);
```

**Comment by iamlkj** [comment]
> Good catch, I can see how this would be a problem for animations that reveal a mesh by scaling from 0. We can merge this after the two formatting issues are addressed.

The other formatting issue we can do is the o

**Comment by romainguy** [comment]
Thanks, could you make sure to sign the CLA please? We can't merge your patch until you do that.

**Comment by romainguy** [comment]
I've pushed the change with formatting corrected myself. Thank you for this suggestion!

**Pull request closed without merging**